### PR TITLE
GH-34: Minor improvements and fixes.

### LINKED ISSUE #34: Minor improvements and fixes
- In MultisigWallet.sol `* @param _newWallet Wallt to add.` -- change Wallt to Wallet.
- In MultisigWallet.sol give credit to Gnosis MultiSigWallet implementation.
- In MultisigWallet.sol isTransactionConfirmed should return true if transaction was already executed.
- Update `copyright 2017` to `copyright 2018`
- Abandon the following part in files' headers (explicitly mentioning utility/value chains):
```
    // ----------------------------------------------------------------------------
    // Utility Chain: MultiSigWallet
    //
    // http://www.simpletoken.org/
    //
    // ----------------------------------------------------------------------------
```
- In TokenHolder.sol -- `keyIsActive` and `keyWasAuthorized` are dead-code modifiers; please remove.
- In TokenHolder.sol -- `keyWasNotAuthorized` should be `keyDoesNotExist` or `keyIsNotAuthorized`
- In TokenHolder.sol -- `keyHasNotExpired` is dead code and should be removed.
- In TokenHolder.sol -- correct the documentation of `submitAuthorizeSession`.
- In TokenHolder.sol revokeSession() -- checking the key is not null (`keyIsNotNull(_ephemeralKey)`) is redundant as implied by `keyIsAuthorized`.
- In TokenHolder.sol -- preferred; better name for `processExecutableTransaction`; possible `verifyExecutableTransaction`.
- In TokenHolder.sol -- inline `isEphemeralKeyActive` in `verifyExecutableTransaction` (processExecutableTransaction) (no need for explicit view function); better readability and no need to read from storage twice.
- In TokenHolder.sol -- remove `recoverKey` function and use `ecrecover` directly.
- In TokenHolder.sol -- reference EIP-1077 in comment line on `getMessageHash`.

### PR DESCRIPTION
**Pull request by pgev**
fixes #34

### CODE REVIEW

File: contracts/Internal.sol [hunk 1]
```diff
@@ -13,13 +13,6 @@ pragma solidity ^0.4.23;
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-//
-// ----------------------------------------------------------------------------
-// Utility Chain: Internal Contract
-//
-// http://www.simpletoken.org/
```
Comment (abhayks1) [inline contracts/Internal.sol]: Is this the new format. What is the reason for removing above ?

File: contracts/TokenHolder.sol [hunk 2]
```diff
@@ -139,22 +121,8 @@ contract TokenHolder is MultiSigWallet {
...
-        require(
-            status != AuthorizationStatus.NOT_AUTHORIZED,
-            "Key was not authorized."
-        );
-        _;
-    }
-
     /** Requires that key was not authorized. */
-    modifier keyWasNotAuthorized(address _key)
+    modifier keyDoesNotExist(address _key)
```
Comment (abhayks1) [inline contracts/TokenHolder.sol]: Better name can be: **keyShouldNotExist** 
Aligns better with submitAuthorizeSession

File: contracts/TokenHolder.sol [hunk 3]
```diff
@@ -342,7 +297,7 @@ contract TokenHolder is MultiSigWallet {
     {
         bytes32 messageHash = bytes32(0);
         address ephemeralKey = address(0);
-        (messageHash, ephemeralKey) = processExecutableTransaction(
+        (messageHash, ephemeralKey) = verifyExecutableTransaction(
```
Comment (abhayks1) [inline contracts/TokenHolder.sol]: verifyExecutableTransaction is making more sense.

File: contracts/TokenHolder.sol [hunk 4]
```diff
@@ -392,34 +347,15 @@ contract TokenHolder is MultiSigWallet {
...
 
-    /**
-     * @notice Checks if the specified key is authorized and non-expired.
-     *
-     * @param _ephemeralKey Key to check.
-     *
-     * @return True if the key is currently authorized and has not expired,
-     *         otherwise false.
-     */
-    function isEphemeralKeyActive(address _ephemeralKey)
```
Comment (abhayks1) [inline contracts/TokenHolder.sol]: We should keep this method as it checks for all the conditions for key to be active and returns true.
Callers can rely on this method to check if key is active or not.
OpenST.js team has already integrated with this method. cc: @kedarchandrayan

File: contracts/MultiSigWallet.sol [hunk 5]
```diff
@@ -590,21 +587,27 @@ contract MultiSigWallet {
...
      */
     function isTransactionConfirmed(uint256 _transactionID)
         internal
         view
         transactionExists(_transactionID)
         returns (bool)
     {
+        if (transactions[_transactionID].executed) {
+            return true;
+        }
```
Comment (abhayks1) [inline contracts/MultiSigWallet.sol]: With above change method can be made public now. Caller can call this method to check if transaction is confirmed or not. Irrespective of transaction is executed or not,  isTransactionConfirmed will work properly.

File: contracts/TokenHolder.sol (hunk 2 above)
Comment (benjaminbollen) [reply contracts/TokenHolder.sol]: `keyDoesNotExist` is clear; state either is or isn't and a function name therefore wouldn't contain "should"

File: contracts/TokenHolder.sol (hunk 4 above)
Comment (benjaminbollen) [reply contracts/TokenHolder.sol]: ok; I had asked @pgev to inline it for improved efficiency on `executeRule`.

We can either keep the function duplicated, but rather we can also perform these checks in openst.js on the public mapping `ephemeralKeys` that is already exposed

File: contracts/MultiSigWallet.sol (hunk 5 above)
Comment (pgev) [reply contracts/MultiSigWallet.sol]: We can have for now it public.

File: contracts/TokenHolder.sol (hunk 4 above)
Comment (pgev) [reply contracts/TokenHolder.sol]: 1. If we bring the function back, we should use it in contract (no code duplication).
2. However, Ben's point is valid, as openst.js can build the function in JS layer.

File: contracts/Internal.sol (hunk 1 above)
Comment (abhayks1) [reply contracts/Internal.sol]: Thank you for the details. I will also follow this for the new contracts.

**Comment by abhayks1** [comment]
@pgev There are merge conflicts. Github not allowing to approve and merge.

**Pull request merged**